lib/internal/process.js

```diff
@@ -283,6 +283,53 @@ function setupUncaughtExceptionCapture(exceptionHandlerState) {
   };
 }
 
+function setupCompareVersion() {
+  const {
+    majorVersion,
+    minorVersion,
+    patchVersion,
+  } = process.release;
+  const prereleaseTag = process.release.prereleaseTag.replace(/^-/, '');
+
+  process.release.compareVersion =
+    (major, minor, patch, prerelease = false) => {
+      if (typeof major === 'string')
+        [major, minor, patch] = major.split('.');
+      if (/-/.test(patch))
+        [patch, prerelease] = patch.split('-');
+
+      major = +major;
+      minor = +minor;
+      patch = +patch;
+      if (/^-/.test(prerelease))
+        prerelease = prerelease.slice(1);
```

Comment:
I don't think this is correct. As far as I understand the specification, `1.1.0--` is a valid version, where `{ major: 1, minor: 1, patch: 0, prerelease: '-' }`.